### ISSUE TITLE
After installing redmine_omniauth_saml plugin bundle is broken

### ISSUE THREAD

**Issue by markwintch** [bundler/bundler#5968]
<script src="https://gist.github.com/markwintch/7f4fa65cb5035b10faa1b033b2cfcfcb.js"></script>
["https://gist.github.com/markwintch/7f4fa65cb5035b10faa1b033b2cfcfcb.js"](url)

PS C:\inetpub\wwwroot\redmine> bundle install --without development test rmagick
Fetching gem metadata from https://rubygems.org/...........
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
--- ERROR REPORT TEMPLATE -------------------------------------------------------
# Error Report

## Questions

Please fill out answers to these questions, it'll help us figure out
why things are going wrong.

- **What did you do?**

  I ran the command `C:/Ruby23-x64/bin/bundle install --without development test rmagick`

- **What did you expect to happen?**

  I expected Bundler to...

- **What happened instead?**

  Instead, what happened was...

- **Have you tried any solutions posted on similar issues in our issue tracker, stack overflow, or google?**

  I tried...

- **Have you read our issues document, https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md?**

  ...

## Backtrace

```
Errno::ENOENT: No such file or directory - git cat-file -e 93d1dd9526e13cf7e999e29c10c70efa65be3c25
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:162:in ``'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:162:in `block (2 levels) in
git'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:239:in `block in capture_and
_filter_stderr'
  C:/Ruby23-x64/lib/ruby/2.3.0/tempfile.rb:295:in `open'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:237:in `capture_and_filter_s
tderr'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:162:in `block in git'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/shared_helpers.rb:73:in `with_clean_git_env'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:161:in `git'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:172:in `block in has_revisio
n_cached?'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/shared_helpers.rb:55:in `chdir'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/shared_helpers.rb:55:in `block in chdir'
  C:/Ruby23-x64/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/shared_helpers.rb:54:in `chdir'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:220:in `in_path'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:172:in `has_revision_cached?
'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git/git_proxy.rb:94:in `checkout'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git.rb:306:in `fetch'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/source/git.rb:160:in `specs'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/lazy_specification.rb:75:in `__materialize__'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/spec_set.rb:84:in `block in materialize'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/spec_set.rb:81:in `map!'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/spec_set.rb:81:in `materialize'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/definition.rb:159:in `specs'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/definition.rb:147:in `resolve_remotely!'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/installer.rb:226:in `resolve_if_need'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/installer.rb:78:in `run'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/installer.rb:24:in `install'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli/install.rb:68:in `run'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:188:in `block in install'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/settings.rb:92:in `temporary'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:187:in `install'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_co
mmand'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:22:in `dispatch'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:13:in `start'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/exe/bundle:30:in `block in <top (required)>'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
  C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/exe/bundle:22:in `<top (required)>'
  C:/Ruby23-x64/bin/bundle:22:in `load'
  C:/Ruby23-x64/bin/bundle:22:in `<main>'
```

## Environment

```
Bundler   1.15.4
Rubygems  2.5.2
Ruby      2.3.3p222 (2016-11-21 revision 56859) [x64-mingw32]
GEM_HOME  C:/Ruby23-x64/lib/ruby/gems/2.3.0
GEM_PATH  C:/Ruby23-x64/lib/ruby/gems/2.3.0;C:/Users/mwintch/.gem/ruby/2.3.0
Git       not installed
Platform  x64-mingw32
OpenSSL   OpenSSL 1.0.2j  26 Sep 2016
```

## Bundler settings

```
without
  Set for your local app (C:/inetpub/wwwroot/redmine/.bundle/config): "development:test:rmagick:trace"
```

### Gemfile

```ruby
source 'https://rubygems.org'

if Gem::Version.new(Bundler::VERSION) < Gem::Version.new('1.5.0')
  abort "Redmine requires Bundler 1.5.0 or higher (you're using #{Bundler::VERSION}).\nPlease update with 'gem update bu
ndler'."
end

gem "rails", "4.2.8"
gem "addressable", "2.4.0" if RUBY_VERSION < "2.0"
gem "jquery-rails", "~> 3.1.4"
gem "coderay", "~> 1.1.1"
gem "request_store", "1.0.5"
gem "mime-types", (RUBY_VERSION >= "2.0" ? "~> 3.0" : "~> 2.99")
gem "protected_attributes"
gem "actionpack-xml_parser"
gem "roadie-rails", "~> 1.1.1"
gem "roadie", "~> 3.2.1"
gem "mimemagic"

gem "nokogiri", (RUBY_VERSION >= "2.1" ? "~> 1.7.2" : "~> 1.6.8")
gem "i18n", "~> 0.7.0"
gem "ffi", "1.9.14", :platforms => :mingw if RUBY_VERSION < "2.0"

# Request at least rails-html-sanitizer 1.0.3 because of security advisories
gem "rails-html-sanitizer", ">= 1.0.3"

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :x64_mingw, :mswin]
gem "rbpdf", "~> 1.19.2"

# Optional gem for LDAP authentication
group :ldap do
  gem "net-ldap", "~> 0.12.0"
end

# Optional gem for OpenID authentication
group :openid do
  gem "ruby-openid", "~> 2.3.0", :require => "openid"
  gem "rack-openid"
end

platforms :mri, :mingw, :x64_mingw do
  # Optional gem for exporting the gantt to a PNG file, not supported with jruby
  group :rmagick do
    gem "rmagick", ">= 2.14.0"
  end

  # Optional Markdown support, not for JRuby
  group :markdown do
    gem "redcarpet", "~> 3.4.0"
  end
end

# Include database gems for the adapters found in the database
# configuration file
require 'erb'
require 'yaml'
database_file = File.join(File.dirname(__FILE__), "config/database.yml")
if File.exist?(database_file)
  database_config = YAML::load(ERB.new(IO.read(database_file)).result)
  adapters = database_config.values.map {|c| c['adapter']}.compact.uniq
  if adapters.any?
    adapters.each do |adapter|
      case adapter
      when 'mysql2'
gem "mysql2", "~> 0.4.6", :platforms => [:mri, :mingw, :x64_mingw]
      when /postgresql/
gem "pg", "~> 0.18.1", :platforms => [:mri, :mingw, :x64_mingw]
      when /sqlite3/
gem "sqlite3", (RUBY_VERSION < "2.0" && RUBY_PLATFORM =~ /mingw/ ? "1.3.12" : "~>1.3.12"),
               :platforms => [:mri, :mingw, :x64_mingw]
      when /sqlserver/
gem "tiny_tds", (RUBY_VERSION >= "2.0" ? "~> 1.2.0" : "~> 0.7.0"), :platforms => [:mri, :mingw, :x64_mingw]
gem "activerecord-sqlserver-adapter", :platforms => [:mri, :mingw, :x64_mingw]
      else
warn("Unknown database adapter `#{adapter}` found in config/database.yml, use Gemfile.local to load your own database ge
ms")
      end
    end
  else
    warn("No adapter found in config/database.yml, please configure it first")
  end
else
  warn("Please configure your config/database.yml first")
end

group :development do
  gem "rdoc", "~> 4.3"
  gem "yard"
end

group :test do
  gem "minitest"
  gem "rails-dom-testing"
  gem "mocha"
  gem "simplecov", "~> 0.9.1", :require => false
  # TODO: remove this after upgrading to Rails 5
  gem "test_after_commit", "~> 0.4.2"
  # For running UI tests
  gem "capybara"
  gem "selenium-webdriver", "~> 2.53.4"
end

local_gemfile = File.join(File.dirname(__FILE__), "Gemfile.local")
if File.exists?(local_gemfile)
  eval_gemfile local_gemfile
end

# Load plugins' Gemfiles
Dir.glob File.expand_path("../plugins/*/{Gemfile,PluginGemfile}", __FILE__) do |file|
  eval_gemfile file
end
```

### Gemfile.lock

```
GIT
  remote: https://github.com/a-ono/rich.git
  revision: 93d1dd9526e13cf7e999e29c10c70efa65be3c25
  specs:
    rich (1.4.8)
      jquery-rails
      kaminari
      mime-types
      rack-raw-upload
      rails (>= 3.2.0)
      sass-rails

GEM
  remote: https://rubygems.org/
  specs:
    actionmailer (4.2.8)
      actionpack (= 4.2.8)
      actionview (= 4.2.8)
      activejob (= 4.2.8)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 1.0, >= 1.0.5)
    actionpack (4.2.8)
      actionview (= 4.2.8)
      activesupport (= 4.2.8)
      rack (~> 1.6)
      rack-test (~> 0.6.2)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionpack-xml_parser (1.0.2)
      actionpack (>= 4.0.0, < 5)
    actionview (4.2.8)
      activesupport (= 4.2.8)
      builder (~> 3.1)
      erubis (~> 2.7.0)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    activejob (4.2.8)
      activesupport (= 4.2.8)
      globalid (>= 0.3.0)
    activemodel (4.2.8)
      activesupport (= 4.2.8)
      builder (~> 3.1)
    activerecord (4.2.8)
      activemodel (= 4.2.8)
      activesupport (= 4.2.8)
      arel (~> 6.0)
    activerecord-sqlserver-adapter (4.2.18)
      activerecord (~> 4.2.1)
    activesupport (4.2.8)
      i18n (~> 0.7)
      minitest (~> 5.1)
      thread_safe (~> 0.3, >= 0.3.4)
      tzinfo (~> 1.1)
    addressable (2.5.1)
      public_suffix (~> 2.0, >= 2.0.2)
    arel (6.0.4)
    builder (3.2.3)
    capybara (2.15.1)
      addressable
      mini_mime (>= 0.1.3)
      nokogiri (>= 1.3.3)
      rack (>= 1.0.0)
      rack-test (>= 0.5.4)
      xpath (~> 2.0)
    childprocess (0.7.1)
      ffi (~> 1.0, >= 1.0.11)
    climate_control (0.2.0)
    cocaine (0.5.8)
      climate_control (>= 0.0.3, < 1.0)
    coderay (1.1.1)
    concurrent-ruby (1.0.5)
    css_parser (1.5.0)
      addressable
    daemons (1.2.4)
    docile (1.1.5)
    erubis (2.7.0)
    eventmachine (1.2.5-x64-mingw32)
    ffi (1.9.18-x64-mingw32)
    globalid (0.4.0)
      activesupport (>= 4.2.0)
    hashie (3.5.6)
    htmlentities (4.3.4)
    i18n (0.7.0)
    jquery-rails (3.1.4)
      railties (>= 3.0, < 5.0)
      thor (>= 0.14, < 2.0)
    kaminari (1.0.1)
      activesupport (>= 4.1.0)
      kaminari-actionview (= 1.0.1)
      kaminari-activerecord (= 1.0.1)
      kaminari-core (= 1.0.1)
    kaminari-actionview (1.0.1)
      actionview
      kaminari-core (= 1.0.1)
    kaminari-activerecord (1.0.1)
      activerecord
      kaminari-core (= 1.0.1)
    kaminari-core (1.0.1)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    macaddr (1.7.1)
      systemu (~> 2.6.2)
    mail (2.6.6)
      mime-types (>= 1.16, < 4)
    metaclass (0.0.4)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mimemagic (0.3.2)
    mini_mime (0.1.4)
    mini_portile2 (2.1.0)
    minitest (5.10.3)
    mocha (1.2.1)
      metaclass (~> 0.0.1)
    multi_json (1.12.1)
    net-ldap (0.12.1)
    nokogiri (1.7.2-x64-mingw32)
      mini_portile2 (~> 2.1.0)
    omniauth (1.2.2)
      hashie (>= 1.2, < 4)
      rack (~> 1.0)
    omniauth-saml-cespi (1.3.1)
      omniauth (~> 1.1)
      ruby-saml (~> 0.9)
    pandoc-ruby (2.0.1)
    paperclip (4.2.4)
      activemodel (>= 3.2.0)
      activesupport (>= 3.2.0)
      cocaine (~> 0.5.5)
      mime-types
    protected_attributes (1.1.4)
      activemodel (>= 4.0.1, < 5.0)
    public_suffix (2.0.5)
    rack (1.6.8)
    rack-openid (1.4.2)
      rack (>= 1.1.0)
      ruby-openid (>= 2.1.8)
    rack-raw-upload (1.1.1)
      multi_json
    rack-test (0.6.3)
      rack (>= 1.0)
    rails (4.2.8)
      actionmailer (= 4.2.8)
      actionpack (= 4.2.8)
      actionview (= 4.2.8)
      activejob (= 4.2.8)
      activemodel (= 4.2.8)
      activerecord (= 4.2.8)
      activesupport (= 4.2.8)
      bundler (>= 1.3.0, < 2.0)
      railties (= 4.2.8)
      sprockets-rails
    rails-deprecated_sanitizer (1.0.3)
      activesupport (>= 4.2.0.alpha)
    rails-dom-testing (1.0.8)
      activesupport (>= 4.2.0.beta, < 5.0)
      nokogiri (~> 1.6)
      rails-deprecated_sanitizer (>= 1.0.1)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    railties (4.2.8)
      actionpack (= 4.2.8)
      activesupport (= 4.2.8)
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rake (12.0.0)
    rb-fsevent (0.10.2)
    rb-inotify (0.9.10)
      ffi (>= 0.5.0, < 2)
    rb-readline (0.5.5)
    rbpdf (1.19.2)
      htmlentities
      rbpdf-font (~> 1.19.0)
    rbpdf-font (1.19.1)
    rdoc (4.3.0)
    redcarpet (3.4.0)
    request_store (1.0.5)
    rmagick (2.16.0)
    roadie (3.2.2)
      css_parser (~> 1.4)
      nokogiri (~> 1.5)
    roadie-rails (1.1.1)
      railties (>= 3.0, < 5.1)
      roadie (~> 3.1)
    ruby-openid (2.3.0)
    ruby-saml (0.9.2)
      nokogiri (>= 1.5.10)
      uuid (~> 2.3)
    rubyzip (1.2.1)
    sass (3.5.1)
      sass-listen (~> 4.0.0)
    sass-listen (4.0.0)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
    sass-rails (5.0.6)
      railties (>= 4.0.0, < 6)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
    selenium-webdriver (2.53.4)
      childprocess (~> 0.5)
      rubyzip (~> 1.0)
      websocket (~> 1.0)
    simplecov (0.9.2)
      docile (~> 1.1.0)
      multi_json (~> 1.0)
      simplecov-html (~> 0.9.0)
    simplecov-html (0.9.0)
    sprockets (3.7.1)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (2.3.3)
      actionpack (>= 3.0)
      activesupport (>= 3.0)
      sprockets (>= 2.8, < 4.0)
    systemu (2.6.5)
    test_after_commit (0.4.2)
      activerecord (>= 3.2)
    thin (1.7.2)
      daemons (~> 1.0, >= 1.0.9)
      eventmachine (~> 1.0, >= 1.0.4)
      rack (>= 1, < 3)
    thor (0.20.0)
    thread_safe (0.3.6)
    tilt (2.0.8)
    tiny_tds (1.2.0-x64-mingw32)
      mini_portile2 (~> 2.0)
    tzinfo (1.2.3)
      thread_safe (~> 0.1)
    tzinfo-data (1.2017.2)
      tzinfo (>= 1.0.0)
    uuid (2.3.8)
      macaddr (~> 1.0)
    websocket (1.2.4)
    xpath (2.1.0)
      nokogiri (~> 1.3)
    yard (0.9.9)

PLATFORMS
  x64-mingw32

DEPENDENCIES
  actionpack-xml_parser
  activerecord-sqlserver-adapter
  capybara
  coderay (~> 1.1.1)
  htmlentities
  i18n (~> 0.7.0)
  imagemagick
  jquery-rails (~> 3.1.4)
  kaminari
  mime-types (~> 3.0)
  mimemagic
  minitest
  mocha
  net-ldap (~> 0.12.0)
  nokogiri (~> 1.7.2)
  omniauth (>= 1.2.2)
  omniauth-saml-cespi (~> 1.3.1)
  pandoc-ruby
  paperclip (~> 4.2.1)
  protected_attributes
  rack-openid
  rails (= 4.2.8)
  rails-dom-testing
  rails-html-sanitizer (>= 1.0.3)
  rb-readline
  rbpdf (~> 1.19.2)
  rdoc (~> 4.3)
  redcarpet (~> 3.4.0)
  request_store (= 1.0.5)
  rich!
  rmagick (>= 2.14.0)
  roadie (~> 3.2.1)
  roadie-rails (~> 1.1.1)
  ruby-openid (~> 2.3.0)
  ruby-saml (~> 0.9.2)
  selenium-webdriver (~> 2.53.4)
  simplecov (~> 0.9.1)
  sprockets-rails (< 3.0.0)
  test_after_commit (~> 0.4.2)
  thin
  tiny_tds (~> 1.2.0)
  tzinfo-data
  yard

BUNDLED WITH
   1.15.3
```

--- TEMPLATE END ----------------------------------------------------------------

Unfortunately, an unexpected error occurred, and Bundler cannot continue.

First, try this link to see if there are any existing issue reports for this error:
https://github.com/bundler/bundler/search?q=No+such+file+or+directory+-+git+cat-file+-e+93d1dd9526e13cf7e999e29c10c70efa
65be3c25&type=Issues

If there aren't any reports for this error yet, please create copy and paste the report template above into a new issue.
 Don't forget to anonymize any private data! The new issue form is located at:
https://github.com/bundler/bundler/issues/new
PS C:\inetpub\wwwroot\redmine>

**Comment by segiddins** [comment]
You need to install git in order to use git gems

**Comment by markwintch** [comment]
Thank you, I am not a developer and Git as well as Github are pretty foreign concepts to me.

I had issues with the other plugin we use CK Editor.  It required GitHub to be installed and it was looking for a specific folder as well.  The workaround I use is that I install Github on the system and I have saved a Poshgit folder from a previous working installation.  

When I run the .\shell.ps1 command in my local appdata directory it fails the first time because it is looking for a specific poshgit folder.  So, my workaround has been to copy over the poshgit I have saved previously and rename it to whatever name I am given in the error message.  This is not pretty and there are likely better ways of doing things, but it is a repeatable process that works for me to get CK Editor installed.

I say all of that because I wonder if I don't have Git installed.  Is GitHub the same thing?  I do not know where to find the folder that this plugin install is looking for.

I appreciate any guidance.

Thank you
